server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/JMAPApiRoutes.scala

```diff
@@ -101,21 +101,32 @@ class JMAPApiRoutes (val authenticator: Authenticator,
 
   private def process(requestObject: RequestObject,
                       httpServerResponse: HttpServerResponse,
-                      mailboxSession: MailboxSession): SMono[Void] =
-    requestObject
-      .methodCalls
-      .map(invocation => this.processMethodWithMatchName(invocation, mailboxSession))
-      .foldLeft(SFlux.empty[Invocation]) { (flux: SFlux[Invocation], mono: SMono[Invocation]) => flux.mergeWith(mono) }
-      .collectSeq()
-      .flatMap((invocations: Seq[Invocation]) =>
-        SMono.fromPublisher(httpServerResponse.status(OK)
-          .header(CONTENT_TYPE, JSON_CONTENT_TYPE)
-          .sendString(
-            SMono.fromCallable(() =>
-              serializer.serialize(ResponseObject(ResponseObject.SESSION_STATE, invocations)).toString),
-            StandardCharsets.UTF_8
-          ).`then`())
-      )
+                      mailboxSession: MailboxSession): SMono[Void] = {
+    DefaultCapabilities.validate(requestObject.using.toSet) match {
+      case true =>     requestObject
+        .methodCalls
+        .map(invocation => this.processMethodWithMatchName(invocation, mailboxSession))
+        .foldLeft(SFlux.empty[Invocation]) { (flux: SFlux[Invocation], mono: SMono[Invocation]) => flux.mergeWith(mono) }
+        .collectSeq()
+        .flatMap((invocations: Seq[Invocation]) =>
+          SMono.fromPublisher(httpServerResponse.status(OK)
+            .header(CONTENT_TYPE, JSON_CONTENT_TYPE)
+            .sendString(
+              SMono.fromCallable(() =>
+                serializer.serialize(ResponseObject(ResponseObject.SESSION_STATE, invocations)).toString),
+              StandardCharsets.UTF_8
+            ).`then`())
+        )
+      case false =>
+        SMono.justOrEmpty(methodsByName.get(invocation.methodName))
+          .flatMap(method => SMono.fromPublisher(method.process(invocation, mailboxSession)))
```

Comment:
Why do we keep trying processing the method if it exists in the case the capabilities are missing from the request?

Reply:
yeah we just commited this wip commit at the end of the day :)